scalalib/src/mill/scalalib/spotless/SpotlessWorker.scala

```diff
@@ -0,0 +1,139 @@
+package mill.scalalib.spotless
+
+import mill._
+import mill.define.{Discover, ExternalModule, Worker}
+import mill.api.{Ctx, PathRef, Result}
+import com.diffplug.spotless.{Formatter, FormatExceptionPolicyStrict}
+import com.diffplug.spotless.generic.LicenseHeaderStep
+import com.diffplug.spotless.java.GoogleJavaFormatStep
+import com.diffplug.spotless.scala.ScalafmtStep
+import com.diffplug.spotless.kotlin.KtlintStep
+import com.typesafe.config.ConfigFactory
+
+import scala.collection.mutable
+import scala.util.control.NonFatal
+
+object SpotlessWorkerModule extends ExternalModule {
+  def worker: Worker[SpotlessWorker] = T.worker { new SpotlessWorker() }
+
+  lazy val millDiscover: Discover = Discover[this.type]
+}
+
+private[spotless] class SpotlessWorker extends AutoCloseable {
```

Comment:
This class should reside in a separate Scala module, so we can load it at runtime in an isolated classloader. We don't want to load the full spotless classpath into all Mill projects.